Applications/Spire/Source/Ui/ColorSelectorHexInputWidget.cpp

```diff
@@ -70,28 +68,30 @@ connection ColorSelectorHexInputWidget::connect_color_signal(
 
 bool ColorSelectorHexInputWidget::eventFilter(QObject* watched,
     QEvent* event) {
-  if(watched == m_line_edit && event->type() == QEvent::KeyPress) {
+  if(watched == m_text_input && event->type() == QEvent::KeyPress) {
     auto e = static_cast<QKeyEvent*>(event);
-    if(e->key() == Qt::Key_Enter || e->key() == Qt::Key_Return) {
-      auto text = m_line_edit->text();
-      if(text.length() == 6 || text.length() == 3) {
-        auto color = QColor();
-        color.setNamedColor(QString("#%1").arg(m_line_edit->text()));
-        m_color_name = color_name(color);
-        m_color_signal(color);
-        if(text.length() == 3) {
-          m_line_edit->setText(m_color_name);
+    switch(e->key()) {
```

Comment:
Use `if` instead as there is only a single branch here.
Normally, one would opt for `switch` only if there are more than two different branches (or expected to be in the future)